tests/backuptestTriggers.go

```diff
@@ -0,0 +1,628 @@
+		RestoreMap:  make(map[string]string),
+		RestoreName: "",
+	}
+
+	var applicationData = ApplicationData{
+		SchedulerContext: make([]*scheduler.Context, 0),
+	}
+
+	var longevityStruct = PxBackupLongevity{
+		CustomData:      &customData,
+		ApplicationData: &applicationData,
+		BackupData:      &backupData,
+		RestoreData:     &restoreData,
+	}
+
+	return longevityStruct
+}
+
+func GetLongevityEventResponse() EventResponse {
+	var someOtherVar = make(map[string]*EventBuilderResponse)
+
+	var eventResponse = EventResponse{
+		EventBuilders: someOtherVar,
+	}
+
+	return eventResponse
+}
+
+// All longevity utilities
+
+// ...
+
+func GetRandomNamespacesForBackup() []string {
+	var allNamespacesForBackupMap = make(map[string]bool)
+	var allNamepsacesForBackup []string
+	rand.Seed(time.Now().Unix()) // initialize global pseudo random generator
+
+	numberOfNamespaces := rand.Intn(len(LongevityAllNamespaces))
+
+	for i := 0; i <= numberOfNamespaces; i++ {
+		allNamespacesForBackupMap[LongevityAllNamespaces[rand.Intn(len(LongevityAllNamespaces))]] = true
+	}
+
+	for namespaceName := range allNamespacesForBackupMap {
+		allNamepsacesForBackup = append(allNamepsacesForBackup, namespaceName)
+	}
+
+	log.Infof("Namespaces selected for restore - [%v]", allNamepsacesForBackup)
+
+	return allNamepsacesForBackup
+}
+
+func GetRandomBackupForRestore() string {
+	rand.Seed(time.Now().Unix()) // initialize global pseudo random generator
+
+	return LongevityAllBackupNames[rand.Intn(len(LongevityAllBackupNames))]
+}
+
+func UpdateEventResponse(eventResponse *EventResponse) {
+	for _, builderResponse := range eventResponse.EventBuilders {
+		eventResponse.TimeTakenInMinutes += builderResponse.TimeTakenInMinutes
+		if builderResponse.Error != nil {
+			eventResponse.Errors = append(eventResponse.Errors, builderResponse.Error)
+		}
+		eventResponse.HighlightEvents = append(eventResponse.HighlightEvents, builderResponse.HighlightEvent)
+	}
+	if eventResponse.Errors != nil {
+		eventResponse.Status = false
+	} else {
+		eventResponse.Status = true
+	}
+	// LogEventData(eventResponse)
+}
+
+// All longevity events
+
+// Event to schedule apps on cluster
+func eventScheduleApps(inputsForEventBuilder *PxBackupLongevity) (error, string, EventData) {
+	defer GinkgoRecover()
+	eventData := &EventData{}
+
+	var scheduledAppContexts = make([]*scheduler.Context, 0)
+	var bkpNamespaces = make([]string, 0)
+
+	for i := 0; i < Inst().GlobalScaleFactor; i++ {
+		taskName := fmt.Sprintf("%s-%d-%s", TaskNamePrefix, i, RandomString(5))
+		appContexts := ScheduleApplications(taskName)
+		for _, ctx := range appContexts {
+			ctx.ReadinessTimeout = AppReadinessTimeout
+			namespace := GetAppNamespace(ctx, taskName)
+			bkpNamespaces = append(bkpNamespaces, namespace)
+			scheduledAppContexts = append(scheduledAppContexts, ctx)
+		}
+	}
+
+	eventData.BackupNamespaces = bkpNamespaces
+	eventData.SchedulerContext = scheduledAppContexts
+
+	return nil, "", *eventData
```

Comment:
remove the empty string return if not required

Reply:
The empty string is required to tract any abnormalities e.g. if an event takes more time than expected, or in case of partial success instead of success, we are not using it for now but it will be used in future enhancements.